### ISSUE TITLE
jdk-17+35 crashes with JFileChooser on Win10

### ISSUE THREAD

**Issue by WLenhard** [adoptium/adoptium-support#385]
## Summary
Invoking JFileChooser leads to crash outside the VM in case, FlatLaF is used as the look and feel.

## Steps to reproduce
Minimal code to reproduce:

```
import javax.swing.JFileChooser;
import javax.swing.SwingUtilities;

import com.formdev.flatlaf.FlatLightLaf;

public class FileChooserTest {

	public static void main(String[] args) {
		
		FlatLightLaf.setup();
		
		SwingUtilities.invokeLater(new Runnable() {
			@Override
			public void run() {
				try {
					JFileChooser chooser = new JFileChooser();
					chooser.showSaveDialog(null);
					String file = chooser.getSelectedFile().getName();
					System.out.println(file);
				} catch (Exception e) {
					e.printStackTrace();
				}

			}
		});
	}
}
```

## Expected results
Show JFileChooser

## Actual results
"An unrecoverable stack overflow has occurred.
A fatal error has been detected by the Java Runtime Environment:
EXCEPTION_STACK_OVERFLOW (0xc00000fd) at pc=0x5e141119, pid=11772, tid=12752

JRE version: OpenJDK Runtime Environment Temurin-17+35 (17.0+35) (build 17+35)
Java VM: OpenJDK Client VM Temurin-17+35 (17+35, mixed mode, emulated-client, serial gc, windows-x86)
Problematic frame:
C  [awt.dll+0xd1119]

No core dump will be written. Minidumps are not enabled by default on client versions of Windows

An error report file with more information is saved as:
... \hs_err_pid11772.log

If you would like to submit a bug report, please visit:
https://github.com/adoptium/adoptium-support/issues
The crash happened outside the Java Virtual Machine in native code.
See problematic frame for where to report the bug."

## Triaging info
Java version:
OpenJDK Client VM Temurin-17+35 (17+35, mixed mode, emulated-client, serial gc, windows-x86)

What is your operating system and platform?
Windows 10 Enterprise, 19042.1288

How did you install Java?
Installer from Adoptium (MSI)

Did it work before?
Yes. No problem with a jdk-16 or earlier.

Did you test with other Java versions?
Yes. OpenJDK Runtime Environment AdoptOpenJDK-16.0.1+9 (build 16.0.1+9) works fine

**Comment by DevCharly** [comment]
It still crashes on Windows in 17.0.1 32bit.
Works in 16.0.2 32bit.

It is not related to FlatLaf. Occurs also in Windows L&F.
It is a bug in OpenJDK when accessing system icons for the file chooser.

There was a change in this area in JDK 17 to access "large Windows icons", which may be responsible for this crash.
See: https://bugs.openjdk.java.net/browse/JDK-8182043

Maybe this change was never tested on 32 Bit because Oracle distributes only 64 Bit JDKs?

Testcase:

~~~java
import javax.swing.JFileChooser;
import javax.swing.SwingUtilities;
import javax.swing.UIManager;

public class FileChooserTest {
    public static void main(String[] args) {
        try {
            UIManager.setLookAndFeel( UIManager.getSystemLookAndFeelClassName() );
        } catch( Exception ex ) {
            ex.printStackTrace();
        }

        SwingUtilities.invokeLater(new Runnable() {
            @Override
            public void run() {
                try {
                    JFileChooser chooser = new JFileChooser();
                    chooser.showSaveDialog(null);
                    String file = chooser.getSelectedFile().getName();
                    System.out.println(file);
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
        });
    }
}
~~~

**Comment by jerboaa** [comment]
@karianna This might be a good item for your Microsoft friends to look into upstream :) 32-bit Windows issue.

**Comment by karianna** [comment]
> @karianna This might be a good item for your Microsoft friends to look into upstream :) 32-bit Windows issue.

Yes - we have David Grieve working on this internally :-)

**Comment by dsgrieve** [comment]
As a workaround, set the minimum thread stack size, e.g. -Xss1M

I believe the default should be 1M, but it appears to be 320K. Since this is set by the linker, this may just be a build issue.

**Comment by dsgrieve** [comment]
Created https://bugs.openjdk.java.net/browse/JDK-8277299

**Comment by DevCharly** [comment]
I think this can be closed because [JDK-8277299](https://bugs.openjdk.java.net/browse/JDK-8277299) is fixed in Java 18 and backported to Java 17.0.3
